src/store/useCalibrationStore.ts

```diff
@@ -0,0 +1,68 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import distance from '@turf/distance';
+import * as turf from '@turf/helpers';
+import { produce } from 'immer';
+import { create } from 'zustand';
+import { createJSONStorage, persist } from 'zustand/middleware';
+
+import { CalibrationProps, CurrentLocationType } from '@/types/Types';
+
+const INITIAL_CALIBRATION: CalibrationProps = {
+  factors: [],
+  meters: [],
+};
+
+type CalibrationState = {
+  calibration: CalibrationProps;
+  actions: {
+    addCalibration: (
+      start: CurrentLocationType,
+      end: CurrentLocationType,
+      steps: number
+    ) => void;
+    resetCalibrationStore: () => void;
+  };
+};
+
+const defaultCalibrationState: Omit<CalibrationState, 'actions'> = {
+  calibration: INITIAL_CALIBRATION,
+};
+
+export const useCalibrationStore = create<CalibrationState>()(
+  persist(
+    (set) => ({
+      ...defaultCalibrationState,
+      actions: {
+        addCalibration: (start, end, steps) =>
+          set(
+            produce((state) => {
```

Comment:
I didn't mention it in the slides, but linked it on the assignment. There is an `immer` middleware for `zustand` which makes it even easier to use, can you refactor to use it in all stores and remove the `produce` calls?

https://docs.pmnd.rs/zustand/integrations/immer-middleware